keps/sig-node/5067-pod-generation/README.md

```diff
@@ -0,0 +1,990 @@
+relates to.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+Because there is only one singular `status.observedGeneration` in the pod status, only
+one writer can set it. This is consistent with other object types that have
+`status.observedGeneration`, and the expectation is that the primary controller for
+the object sets the field. For a pod, that primary controller is the kubelet, so
+once a pod is bound to a node, we expect that the kubelet on that node is the
+sole writer of `status.observedGeneration`. 
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+Today, it is possible for a client to custom-set a pod's `metadata.generation`, but once
+set, the `metadata.generation` cannot be updated. That means that there is a theoretical 
+possibility for an external client to be setting `metadata.generation` on pod 
+create and depend on that fixed value somehow in its own reconciliation logic.
+
+That said, the `metadata.generation` field is described as ["set by the system and monotonically increasing"](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata),
+and thus it should be clear enough that `metadata.generation` was not intended to be
+used in this way.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+#### API server
+
+For a newly created pod, the API server will set `metadata.generation` to 1. For any updates
+to the [PodSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#podspec-v1-core), the API server will increment `metadata.generation` by 1. The logic to set new pods'
+`metadata.generation` to 1 and to increment `metadata.generation` on update will
+run after all mutating webhooks have finished. 
+
+The pod `metadata.generation` will also continue to be incremented on graceful delete or 
+deferred delete, just as the API server currently does for pods and other 
+objects today.
+
+Pod updates that would not result in `metadata.generation` being incremented include:
+* Changes to metadata. This means that if a Pod uses the downward API to make
```

Comment:
`deletionTimestamp` is in metadata, does the generation depend on it?

Reply:
yeah I forgot to specify here that `deletionTimestamp` is an exception - added that to the body of the kep to clarify